src/utils/uri.ts

```diff
@@ -38,30 +35,32 @@ export class Erc1319URI {
           `Invalid package ID: ${packageId}. URI must include both a package name and version.`,
         );
       } else {
-        return packageId;
+        let [packageName, version] = packageId
+        let namespacedAsset = pathElements.splice(1).join("/")
+        return [packageName, version, namespacedAsset];
       }
     } else {
-      return ['', ''];
+      return ['', '', ''];
     }
   }
 
   static parseChainId(url: URL) {
     // todo: support multiple chain ids
     const chainId = +url.port;
-    if (chainId !== 1) {
+    if (chainId === 1 || chainId === 0) {
```

Comment:
Is `url.port` guaranteed to exist? Just a warning that in JS, `+undefined` returns `NaN`, not zero.